anise/src/math/cartesian.rs

```diff
@@ -123,6 +124,52 @@ impl CartesianState {
         )
     }
 
+    /// Creates a new [CartesianState] for a GPS vehicle position,
+    /// expressed in proposed [Frame] that we augment with appropriate
+    /// [Ellipsoid] shape.
+    ///
+    /// **Units:** km, km, km
+    pub fn from_gps_position(x_km: f64, y_km: f64, z_km: f64, epoch: Epoch, frame: Frame) -> Self {
```

Comment:
ANISE does not to define any frames at all. All of the frames are loaded into the context (the _Almanac_) at runtime and the user fetches the frame that they want that way.

For example:

```
let eme2000 = almanac.frame_from_uid(EARTH);
```

To add new frames, you'll need to define the frame in a  "planetary constants ANISE" file ("PCA"). I still need to write tutorials in Rust, but the simplest method is probably to load the default PCA file (pck11.pca), then create the new frame you want (in this case I'm guessing it's the Earth J2000 (or inertial) frame), and modify the shape. That creation only needs to happen once so you can save the file and reuse it later on any platform (regardless of the byte order).

In practice, it looks something like this (I'm doing this from memory).

```rust
let mut almanac = MetaAlmanac::latest().unwrap();
let mut earth_inertial = almanac.frame_from_uid(EARTH); // Earth is 399, 3 being the 3rd body from the Sun and 99 being the planet itself, 301 would be the first satellite of the Earth i.e. the Moon
earth_inertial.shape = gps_wgs84;

// Doc link: https://docs.rs/anise/latest/anise/structure/dataset/struct.DataSet.html#method.push
almanac.planetary_data.push(earth_inertial, Some(399001), Some("GPS WGS84")).unwrap(); // I'm arbitrarily setting the GPS WGS84 to ID 399001, but feel free to change that. The Moon uses 31000 and single digit increments on top of that
almanac.planetary_data.save_as("pck_with_wgs84.pca").unwrap()
```
The `pck_with_wgs84.pca` file is pretty small, so it can easily be embedded.

Reply:
I can start simple by defining WGS84 (for example) with a macro and not having a custom PCA, but it is one operation we will have to do quite often. So eventually that's what we have to do.
It's pretty cool we can define new datasets

Reply:
I think I'll add an `anise` option to gnss-rs, which is the core basis definitions to all this work, where we can define these datasets